endoreg_db/views/video/video_stream.py

```
@@ -0,0 +1,306 @@
+        
+    Note:
+        Permissions are handled by the calling view, not in this helper function.
+        HTTP 206 Partial Content support is critical for video seeking in browsers.
+    """
+    try:
+        # Determine which file to stream based on file_type
+        if file_type == 'raw':
+            if hasattr(vf, 'active_raw_file') and vf.active_raw_file and hasattr(vf.active_raw_file, 'name'):
+                file_ref = vf.active_raw_file
+            else:
+                raise Http404("No raw video file available for this entry")
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)